widget/controllers/widget.home.controller.js

```diff
@@ -32,13 +32,23 @@
 
         WidgetHome.currentDate = new Date();
         WidgetHome.yesterdayDate = +WidgetHome.currentDate.setDate(WidgetHome.currentDate.getDate() - 1);
+        WidgetHome.todayDate = +(new Date());
         var searchOptions = {
           skip: 0,
           limit: PAGINATION.itemCount,
           filter: {
-            "$or": [{
-              "$json.expiresOn": {$gte: WidgetHome.yesterdayDate}
-            }, {"$json.expiresOn": ""}]
+            "$and": [
+              {
+                "$or": [{
+                  "$json.expiresOn": { $gte: WidgetHome.yesterdayDate }
+                }, { "$json.expiresOn": "" }]
```

Comment:
Why are we filtering empty strings? @nldn 
Same question goes for each filter you added

Reply:
This filters the case if attribute is not set.